src/QAT/qat/purr/compiler/execution.py

```diff
@@ -219,15 +224,20 @@ def _generate_repeat_batches(self, repeats):
         )
         return ([batch_limit] * list_expansion_ratio) + [repeats % batch_limit]
 
-    def execute(self, instructions):
+    def execute(self, instructions) -> Dict[str, Any]:
```

Comment:
This can just be moved up to `InstructionExecutionEngine`, it no longer needs to be abstract.